Analysis/Tasks/vertexerhf.cxx

```diff
@@ -116,7 +116,7 @@ struct DecayVertexBuilder2Prong {
         o2::track::TrackParCov trackparvar1(x1_, alpha1_, arraypar1, covpar1);
         df.setUseAbsDCA(true);
         int nCand = df.process(trackparvar0, trackparvar1);
-        //FIXME: currently filling the table for all dca candidates.
+        df.propagateTracksToVertex();
```

Comment:
You don't need to do this since the fitter is ordered to do this automatically for every successfully fitter vertex:  https://github.com/AliceO2Group/AliceO2/blob/da113c9c9352ef0454515067ab13c646f381142f/Analysis/Tasks/vertexerhf.cxx#L56
Instead you should always check the return value: if the nCand is 0, then no vertex is found and you should not try to retrieve the fitted tracks or vertices. If you want I can add and assert against an attempt to access fit results in case of non-convergence, but usually I let the code to crash in case of improper access....

Reply:
Ciao @shahor02, I read just a part of your message yesterday night. Sorry was a stupid bug I introduced when I removed the loop over the candidates and I forgot to explicitly check for nCand>0. It works fine now. Thanks again. GM